src/plugins/discord/sqliteMirror.js

```diff
@@ -0,0 +1,502 @@
+// @flow
+
+import type Database from "better-sqlite3";
+import stringify from "json-stable-stringify";
+import * as Model from "./models";
+import dedent from "../../util/dedent";
+import * as NullUtil from "../../util/null";
+
+const VERSION = "DISCORD_MIRROR_v1";
+
+/**
+ * Persists a local copy of data from a Discord Guild.
+ * Implements create and read functionality.
+ *
+ * Each Mirror instance is tied to a particular Guild. Trying to use a mirror
+ * for multiple Discord Guilds is not permitted; use separate Mirrors.
+ *
+ * Note that Mirror persists separate Tables for Users and Guild Members.
+ * Members are distinguished by membership in the Guild. Non-Member
+ * Discord Users can participate in a Guild's activity by leaving comments,
+ * reactions, etc. In our model, Members have a property, User, which
+ * represents a full User object. Because of this, we save the User in the
+ * AddMember method.
+ */
+
```

Comment:
Nit: since you've submitted #1747 I thought I would mention this.

A blank line between the comments and declaration should be avoided, when the comment applies to the declaration. I would interpret the blank line as, not directly related to the declaration.

You can see an example of both in `src/backend/loadContext.js`.

Directly commenting on the class:
https://github.com/sourcecred/sourcecred/blob/890222279a1e194f6202cad00ca6dc982dd8e1bd/src/backend/loadContext.js#L40-L52

Not commenting on one property, but a section:
https://github.com/sourcecred/sourcecred/blob/890222279a1e194f6202cad00ca6dc982dd8e1bd/src/backend/loadContext.js#L59-L75

Reply:
Updated.

Reply:
Thanks for the explanation too. Makes sense.